QuietFactory.php

```diff
@@ -49,11 +48,21 @@ public function createItem($name, array $options = array())
         try {
             return $this->innerFactory->createItem($name, $options);
         } catch (RouteNotFoundException $e) {
-            $this->logger->error(sprintf('%s : %s', $name, $e->getMessage()));
+            if (null !== $this->logger) {
+                $this->logger->error(
+                    sprintf('An exception was thrown while creating a menu item called "%s"', $name),
+                    array('exception' => $e)
+                );
+            }
 
             if (!$this->allowEmptyItems) {
                 return null;
             }
+
+            // remove route to create an item without any URL
+            unset($options['route']);
```

Comment:
Hmm, you're correct about the content.

What about reverting this unset and setting linkType to `uri` instead? This means that the route option will be unset by the ContentExtension of this bundle and the content object (if available) will still be stored in the extra's.